lib/node_modules/@stdlib/math/base/special/kernel-tan/src/addon.c

```diff
@@ -29,86 +33,12 @@
 * @return       Node-API value
 */
 static napi_value addon( napi_env env, napi_callback_info info ) {
-	napi_status status;
-
-	// Get callback arguments:
-	size_t argc = 3;
-	napi_value argv[ 3 ];
-	status = napi_get_cb_info( env, info, &argc, argv, NULL, NULL );
-	assert( status == napi_ok );
-
-	// Check whether we were provided the correct number of arguments:
-	if ( argc < 3 ) {
-		status = napi_throw_error( env, NULL, "invalid invocation. Insufficient arguments." );
-		assert( status == napi_ok );
-		return NULL;
-	}
-	if ( argc > 3 ) {
-		status = napi_throw_error( env, NULL, "invalid invocation. Too many arguments." );
-		assert( status == napi_ok );
-		return NULL;
-	}
-
-	napi_valuetype vtype0;
-	status = napi_typeof( env, argv[ 0 ], &vtype0 );
-	assert( status == napi_ok );
-	if ( vtype0 != napi_number ) {
-		status = napi_throw_type_error( env, NULL, "invalid argument. First argument must be a number." );
-		assert( status == napi_ok );
-		return NULL;
-	}
-
-	napi_valuetype vtype1;
-	status = napi_typeof( env, argv[ 1 ], &vtype1 );
-	assert( status == napi_ok );
-	if ( vtype1 != napi_number ) {
-		status = napi_throw_type_error( env, NULL, "invalid argument. Second argument must be a number." );
-		assert( status == napi_ok );
-		return NULL;
-	}
-
-	napi_valuetype vtype2;
-	status = napi_typeof( env, argv[ 2 ], &vtype2 );
-	assert( status == napi_ok );
-	if ( vtype1 != napi_number ) {
-		status = napi_throw_type_error( env, NULL, "invalid argument. third argument must be a number." );
-		assert( status == napi_ok );
-		return NULL;
-	}
-
-	double x;
-	status = napi_get_value_double( env, argv[ 0 ], &x );
-	assert( status == napi_ok );
-
-	double y;
-	status = napi_get_value_double( env, argv[ 1 ], &y );
-	assert( status == napi_ok );
-
-	int32_t k;
-	status = napi_get_value_int32( env, argv[ 2 ], &k );
-	assert( status == napi_ok );
-
-	double out = stdlib_base_kernel_tan( x, y, k );
-
-	napi_value v;
-	status = napi_create_double( env, out, &v );
-	assert( status == napi_ok );
-
-	return v;
-}
-
-/**
-* Initializes a Node-API module.
-*
-* @param env      environment under which the function is invoked
-* @param exports  exports object
-* @return         main export
-*/
-static napi_value init( napi_env env, napi_value exports ) {
-	napi_value fcn;
-	napi_status status = napi_create_function( env, "exports", NAPI_AUTO_LENGTH, addon, NULL, &fcn );
-	assert( status == napi_ok );
-	return fcn;
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_DOUBLE( env, x, argv, 0 );
+	STDLIB_NAPI_ARGV_DOUBLE( env, y, argv, 1 );
+	STDLIB_NAPI_ARGV_INT32( env, k, argv, 2 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, (double)stdlib_base_kernel_tan( x, y, k ), out );
```

Comment:
```suggestion
	STDLIB_NAPI_CREATE_DOUBLE( env, stdlib_base_kernel_tan( x, y, k ), out );
```

`kernel_tan` already returns a `double`.